### PR TITLE
Expands description of ShadowedDeclarationInspection

### LINKED ISSUE #4609: ShadowedDeclarationInspection description text should mention fully qualified identifiers
Currently, the English description is

> Two declarations are in scope and have the same identifier name. This means that only one of them will be available to use.

This is true for *unqualified* identifiers, but certainly not for *fully qualified identifiers*. The description should mention that. (Even if it only serves to have a still-ignorant VBA newbie search for the term...)

This might even be a good place to offer the quickfix mentioned in #4194 (once it's implemented).

### LINKED ISSUE #4614: ShadowedDeclarationInspection severity defaults to 'Do not show'
When I delete my `rubberduck.config` file and start up RD, the severity for ShadowedDeclarationInspection is set to 'Do not show'. I don't think this is intentional.

These inspections are currently (4269-pre) defaulting to 'Do not show':

1. Redundant 'ByRef' modifier
2. Shadowed declaration
3. 'For...Next' loop step is not specified

For 1 and 3, one could argue that not showing them by default is sensible. For 2, I think there's no reason not to show it. (Or am I missing something?)

I tried finding out where this default setting is coming from, but I'm not getting it (yet).

[The CodeInspectionSettings](https://github.com/rubberduck-vba/Rubberduck/blob/406bbb0a7bddc9d438a8eb869f8ca95184dbcc05/Rubberduck.Core/Settings/CodeInspectionSettings.cs#L142) seem to have `CodeInspectionSeverity.Warning` as default value, as does the [InspectionBase](https://github.com/rubberduck-vba/Rubberduck/blob/406bbb0a7bddc9d438a8eb869f8ca95184dbcc05/Rubberduck.CodeAnalysis/Inspections/Abstract/InspectionBase.cs#L47) class.

Are there other places where default settings are set?

### PR DESCRIPTION
**Pull request by Inarion**
Since this is my first ever PR, I'm not (yet) aware of all the things I might be doing wrong. So please enlighten me! :)

Closes #4609
Closes #4614

### CODE REVIEW

**Comment by Inarion** [comment]
Let's keep this open a little longer. There is [a fix for a typo](https://chat.stackexchange.com/transcript/message/47916780#47916780) @comintern suggested I could include in here as well.

Also, #4614 seems like it would fit right in.

**Comment by Inarion** [comment]
No further changes/additions from my side anymore.

Closes #4614

File: Rubberduck.Core/Properties/Settings.settings [hunk 1]
```diff
@@ -231,7 +231,7 @@
               &lt;CodeInspection Name="EmptyModuleInspection" Severity="Hint" InspectionType="MaintainabilityAndReadabilityIssues" /&gt;
               &lt;CodeInspection Name="LineLabelNotUsedInspection" Severity="Warning" InspectionType="CodeQualityIssues" /&gt;
               &lt;CodeInspection Name="IntegerDataTypeInspection" Severity="Hint" InspectionType="CodeQualityIssues" /&gt;
-              &lt;CodeInspection Name="ShadowedDeclarationInspection" Severity="DoNotShow" InspectionType="CodeQualityIssues" /&gt;
+              &lt;CodeInspection Name="ShadowedDeclarationInspection" Severity="Warning" InspectionType="CodeQualityIssues" /&gt;
```
Comment (retailcoder) [inline Rubberduck.Core/Properties/Settings.settings]: Hmm, I would expect `App.config` to mirror that change.

Comment (Inarion) [reply Rubberduck.Core/Properties/Settings.settings]: Ok, so I just wrote a lengthy reply explaining in how many ways I don't understand what you're saying, and justifying what I had to make sure it all worked. Just before posting, it dawned on me that I might have missed one test: Running RD without a config file. Now I see what you mean.

That unfortunately doesn't mean I understand what exact `App.config` you're referring to (there's none in the `.Resources` namespace). And I don't get the how and why. In my very simple mental model of RD's inner workings, it seems like we'd like to have such things stored in only one place.

And if you now tell me to go and read a book about C# software architecture and XAML, I guess you're probably right. ;)

Comment (retailcoder) [reply Rubberduck.Core/Properties/Settings.settings]: Ah, sorry - that would be under `Rubberduck.Core`: https://github.com/rubberduck-vba/Rubberduck/blob/next/Rubberduck.Core/app.config

Comment (MDoerner) [reply Rubberduck.Core/Properties/Settings.settings]: After building, `app.config` should mirror the change.

**Comment by MDoerner** [comment]
Would you like to resolve this conflict yourself or shall I make a PR to your fork? Probably, you just have to take both changes.

**Comment by MDoerner** [comment]
I just had a closer look. It seems as if one of the previous changes has moved the entire code inspections section to the top. You will have to take `next` and reapply your change in the top section.

**Comment by Inarion** [comment]
app.config still doesn't get updated automatically. Not sure why that is. What should I do with it?

**Comment by Inarion** [comment]
@bclothier Regarding your ping in chat: As mentioned above, I'd like to finish this PR, but I don't know what's needed to do so. I was told there should be an automatic update to app.config during the build process, but there isn't. (So my assumption would be that I'm doing something wrong. On the other hand, everything seems to work fine after building.)

AppVeyor seems to fail due to unrelated (?) unit tests failing.

**Comment by bclothier** [comment]
First try and pull from `next` (which has been updated) and verify if you are able to build and run all tests locally. If you need assistance in the particulars, use chat.

**Pull request merged**